docs(web-modeler): add environment variables section to git-sync (#5008)
camunda/camunda-docs: docs/components/modeler/web-modeler/git-sync.md

Creating multiple copies of a process application can complicate navigation and deployment if you have multiple files with the same ID in a project. To avoid this, you can create copies of the process application in different projects.
:::

## Self-Managed environment variables

> The following environment variables can be set to configure the Git sync feature in a self-managed environment:

Reviewer: ```suggestion
You can set the following environment variables to configure Git sync in a Self-Managed environment:
```
Rewording suggestion

| Provider      | Variable                                     | Description                                                                                                                   | Default                     |
| ------------- | -------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------- | --------------------------- |

> | All providers | `CAMUNDA_MODELER_GITSYNC_MAX_FILES`          | Maximum number of allowed files for sync operations.                                                                                    | `50`                        |

Reviewer 2: I think the name should be without underscore (see also https://docs.spring.io/spring-boot/reference/features/external-config.html#features.external-config.typesafe-configuration-properties.relaxed-binding). I've tried both variants and only the one without underscore worked for me (in similar cases in the past Spring accepted both if I remember correctly 🤷🏻).
```suggestion
| All providers | `CAMUNDA_MODELER_GITSYNC_MAXFILES`          | Maximum number of allowed files for sync operations.                                                                                    | `50`                        |
```

> | All providers | `CAMUNDA_MODELER_GITSYNC_MAX_FILES`          | Maximum allowed files for sync operations.                                                                                    | `50`                        |

Reviewer: ```suggestion
| All providers | `CAMUNDA_MODELER_GITSYNC_MAX_FILES`          | Maximum number of allowed files for sync operations.                                                                                    | `50`                        |
```
Add clarification

> | All providers | `CAMUNDA_MODELER_GITSYNC_MAX_IN_MEMORY_SIZE` | Maximum memory size that can be processed by calls to the git provider. This limits the maximum file size that can be synced. | `4MB`                       |

Reviewer 2: Same here.
```suggestion
| All providers | `CAMUNDA_MODELER_GITSYNC_MAXINMEMORYSIZE` | Maximum memory size that can be processed by calls to the git provider. This limits the maximum file size that can be synced. | `4MB`                       |
```
